src/nvim/terminal.c

```diff
@@ -365,6 +365,16 @@ void terminal_resize(Terminal *term, uint16_t width, uint16_t height)
   invalidate_terminal(term, -1, -1);
 }
 
+void terminal_resize_into(Terminal *term, win_T* wp)
+{
+  int width = wp->w_width;
+  if (wp->w_p_nu || wp->w_p_rnu) {
+    width -= wp->w_nrwidth + 1;
+  }
```

Comment:
Is this how it looks in other code that deals with w_nrwidth?

Reply:
This is an example of how `w_nrwidth` is calculated:

https://github.com/neovim/neovim/blob/56eda2aa17c80ba380b606f9466f288fb8162dd3/src/nvim/screen.c#L691

I don't actually see any uses of it besides determining if a redraw needs to happen.

Reply:
Would `win_col_off()` work instead? E.g.

```c
    terminal_resize(buf->terminal,
                    (uint16_t)(MAX(0, curwin->w_width - win_col_off(curwin))),
                    (uint16_t)curwin->w_height)
```

I think this should be done at each call site, instead of adding a new `terminal_resize_into` function. Because many of those call sites are going to be removed later, for other reasons.

Reply:
That's actually how I did it originally. I'll switch back to doing it that way.